### ISSUE TITLE
Running Tomcat under Windows, the Thumbs.db file is not ignored and causes OpenCV error

### ISSUE THREAD

**Issue by Jim-Salmons** [chreul/LAREX#64]
To test this, I added a test folder of the same image files with and without the hidden system Thumbs.db file. The one without the file works as expected. The one with the Thumbs.db file throws this OpenCV error:

`OpenCV Error: Assertion failed (scn == 3 || scn == 4) in cv::cvtColor, file C:\build\2_4_winpack-bindings-win64-vc14-static\opencv\modules\imgproc\src\color.cpp, line 3739
`

and hangs until you leave the folder and go back to the book selection level of the webapp.

BTW, I am running Larex under Windows using the latest OpenCV 2.4.x level (not 3.x) and the latest Windows Tomcat. Also, I have a wonderful dev environment up and running under JetBrains' Intellij IDEA (Ultimate Edition) with build-to-debug workflow working every well. Rather than follow the brief install instructions as written, I used Intellij's Maven project import and ensured that my Project configs were reasonable. The only "gotcha" I hit was to ensure that the OpenCV "path-to-opencv\build\java\x64" was included in Windows' PATH statement as this is what Java under Windows uses for its search paths for Java executables/jars, etc.

The only other new behavior I had to figure out was that, unlike running a simple Python script -- like I have been doing for FactMiners and The Softalk Apple Project metadata "ferret" scripts -- Intellij requires a two-step exit from a debugging session. The first step -- click the "red square" -- stops the program execution as expected. But this red square will be replaced by a dimmed skull which I learned means that the script is dead but Java and Tomcat are still running. A second click on the skull fully kills the debug session and you are ready for coding and rebuilding, etc.

I have attached a couple screenshots that document this issue.

![larex_on_windows_thumb db_file_shows](https://user-images.githubusercontent.com/546022/29006493-8c5f39dc-7aae-11e7-9978-a0bb3e8520dc.png)
![larex_on_windows_thumb db_file_opencv_error](https://user-images.githubusercontent.com/546022/29006496-91764ee2-7aae-11e7-9a51-edfba15732e1.png)

**Comment by Jim-Salmons** [comment]
Hi Christian and Nico,

This issue would appear to come down to either filtering non-renderable files from getting into the Book's pages property in the first place, or checking on a file image's render-ability in the editor panel  when the book is opened or the page changed. As can be seen in my screenshots, the booksidebar handles non-renderable items handed to it by the Book's pages property. 

I think there is less chance for user and/or workflow interaction inconsistencies if the non-renderable files are ignored or filtered out of the returned value of Book.getPages(). 

Note, BTW, that this filtering issue might impact the index numbering that maps files to Larex-rendered pages. So that might be a side-effect to consider when fixing this issue.

Finally, related to this last point about image to page/index numbers, one "scratch your own itch" aspect that I will be looking into is how Larex -- which is already supporting PRImA's PAGE #GTS format -- might also support FactMiners' evolving MAGAZINE #GTS format. For example, looking at our case of digital collections stored at the Internet Archive, we have identified the primacy for fine-grained trace-ablity to the "leaf2ppg" map which is the Internet Archive's unique ID and the serial document's physical PPG or print page number.

In closing, along with optional support of "leaf2ppg" mapping in support of Internet Archive collections, I intend to help with providing the ability for Larex to directly access digital collections at the Internet Archive along the lines of the Python-based proof-of-concept metadata "ferret" scripts we have been working on at FactMiners as part of The Softalk Apple Project.

I look forward to contributing to the Larex project to the extent that optional features to support FactMiners' MAGAZINE #GTS format are consistent with the goals and values of your outstanding project.

Happy, Healthy Vibes,
-- Jim --

P.S. As a further aside... as the bulk of my career was spent as a Smalltalker, I am not a great fan of Java. So another of my personal interests will be in making my optional Larex extension contributions in Kotlin rather than native Java. Along these lines, as mentioned in my original post, I have a really wonderful development environment set up in Windows under JetBrains' Intellij IDEA IDE. This, too, is to further enable my interest in using Neo4j's graph database for MAGAZINE-related extensions related to an integrated complex document structure and content depiction model based on a cidocCRM/FRBRoo/PRESSoo ontological stack. I look forward to working with you and your team.

**Comment by chreul** [comment]
Hello Jim,

thank you for pointing out this issue. As you already mentioned, this happens because the input files aren't filtered. There was a filtering function in the desktop version of the tool but apparently it got lost when we built the new GUI. This issue will be addressed in the very near future, probably while also adding .tif support.

Since supporting additional formats definitely is one of our long term goals any help in this direction is dearly appreciated. We also look forward to working with you. Thank you very much!

Best regards
Christian

**Comment by NesbiDevelopment** [comment]
Greetings,
files in the book folder are now being filtered.
Supported file extensions for images are atm 'png', 'jpeg' and 'jpg'.

Added and merged with cb022da

**Comment by Jim-Salmons** [comment]
Thanks for the update. I hesitated to contribute a fix as I don't yet know your design goals/style architecture-wise, etc. So this will be helpful, getting to see how you folks have implemented the filter.

My thought was to check the OpenCV API to see if there was a call to get the image types it supports,  but a hard-coded filter makes quick-fix sense. If there evolves a configuration file, a simple string w/ file acceptable file extensions might be useful especially if you get to a design architecture which supports image repository plug-ins. I, for example, intend to look at what is needed to directly access jpg/leafs in the collections of magazines at the Internet Archive. I am sure you have or will be developing a "clientele" of Europeana-funded projects that would want such image repository access flexibility.

Another related thought for that configuration file would  be a user settable regex for filtering selections.

Keep up the great work... BTW, a tweet I did during #DH2017 has been getting a Ripple Effect of folks reading through the event stream and doing retweets long after the fact: https://twitter.com/Jim_Salmons/status/895443692969877504.

:-)
